lua/neotest-python/base.lua

```diff
@@ -21,25 +21,50 @@ M.module_exists = function(module, python_command)
   })) == 0
 end
 
+local mem = {}
+
 ---@return string[]
 function M.get_python_command(root)
+  if mem[root] then
+    return mem[root]
+  end
   -- Use activated virtualenv.
   if vim.env.VIRTUAL_ENV then
-    return { Path:new(vim.env.VIRTUAL_ENV, "bin", "python").filename }
+    mem[root] = { Path:new(vim.env.VIRTUAL_ENV, "bin", "python").filename }
+    return mem[root]
   end
 
   for _, pattern in ipairs({ "*", ".*" }) do
     local match = async.fn.glob(Path:new(root or async.fn.getcwd(), pattern, "pyvenv.cfg").filename)
     if match ~= "" then
-      return { (Path:new(match):parent() / "bin" / "python").filename }
+      mem[root] = { (Path:new(match):parent() / "bin" / "python").filename }
+      return mem[root]
     end
   end
 
   if lib.files.exists("Pipfile") then
-    return { "pipenv", "run", "python" }
+    local f = assert(io.popen("pipenv --py 2>/dev/null", "r"))
+    local venv = assert(f:read("*a")):gsub("\n", "")
```

Comment:
Rather than using `io.popen`, let's use the `lib.process.run` function as it runs asynchronously. Doc is here if you don't have type checking on for neotest repo https://github.com/nvim-neotest/neotest/blob/master/lua/neotest/lib/process/init.lua#L14

Reply:
You need to split the args, just pass `{"pipenv", "--py"}`, you don't need the redirection because the run function keeps stdout and stderr separate

Reply:
@rcarriga updated